test/jdk/java/awt/event/MouseEvent/AWTPanelSmoothWheel.java

```diff
@@ -0,0 +1,132 @@
+            panel below the instruction window.<br> <br>
+
+            A hi-res mouse is one which produces MouseWheelEvents having:
+            <pre><b> Math.abs(preciseWheelRotation) &lt; 1. </b></pre><br>
+
+            When preciseWheelRotation adds up, wheelRotation becomes non-zero
+            (can be negative when mouse wheel is scrolled down). <br>
+            You should see many events where the absolute value of
+            preciseWheelRotation &lt; 1 &amp; wheelRotation = 0 followed by
+            an event where wheelRotation != 0 in the logs.<br> <br>
+
+            Check if the test works OK when the mouse wheel is rotated very slow.<br> <br>
+            This is a semi-automated test, if you are using a hi-res mouse and
+            it satisfies the hi-res MouseWheelEvents as described above,
+            the test should automatically pass.<br> <br>
+
+            <hr>
+            PLEASE NOTE:
+            <ul>
+                <li> If you don't see events with preciseWheelRotation &lt; 1,
+                then the mouse doesn't support high-resolution scrolling.</li>
+                <li> A warning is shown if you are not using a hi-res mouse. </li>
+                <li> MouseWheelEvent logs are displayed in the log area
+                for user reference. </li>
+                <li> When mouse is scrolled up, preciseWheelRotation & wheelRotation are positive
+                 and they are negative when scrolled down. </li>
+            </ul>
+            <br>
+            </body>
+            </html>
+            """;
+
+    public static void main(String[] args) throws Exception {
+        PassFailJFrame.builder()
+                .title("Test Instructions")
+                .instructions(INSTRUCTIONS)
+                .rows(30)
+                .columns(54)
+                .testTimeOut(10)
+                .logArea(10)
+                .testUI(AWTPanelSmoothWheel::createUI)
+                .build()
+                .awaitAndCheck();
+    }
+
+    private static Frame createUI() {
+        Frame frame = new Frame("Test Wheel Rotation");
+        Panel panel = new Panel();
+        panel.setBackground(Color.GREEN);
+        panel.addMouseWheelListener(e -> {
+            if (e.getScrollType() == MouseWheelEvent.WHEEL_UNIT_SCROLL) {
+                PassFailJFrame.log("WheelEvent#" + (++wheelEventCount)
+                        + " --- Wheel Rotation: " + e.getWheelRotation()
+                        + " --- Precise Wheel Rotation: "
+                        + String.format("%.2f", e.getPreciseWheelRotation()));
+                if (Math.abs(e.getPreciseWheelRotation()) < 1) {
+                    hiResWheelCount++;
+                }
+                if (wheelEventCount >= 5 && hiResWheelCount == 0) {
+                    PassFailJFrame.log(WARNING_MSG);
+                    JOptionPane.showMessageDialog(frame, WARNING_MSG,
+                            "WARNING", WARNING_MESSAGE);
```

Comment:
```suggestion
                            "WARNING", JOptionPane.WARNING_MESSAGE);
```
I suggest using `JOptionPane.WARNING_MESSAGE` explicitly: there's no confusion between `WARNING_MSG` and `WARNING_MESSAGE` then.

I'd use ‘Warning’ in its title case as opposed to uppercase, taking into account the message itself contains upper case warning.